src/modules/local_position_estimator/BlockLocalPositionEstimator.cpp

```diff
@@ -0,0 +1,962 @@
+		_pub_gpos.get().lon = lon;
+		_pub_gpos.get().alt = alt;
+		_pub_gpos.get().vel_n = _x(X_vx);
+		_pub_gpos.get().vel_e = _x(X_vy);
+		_pub_gpos.get().vel_d = _x(X_vz);
+		_pub_gpos.get().yaw = _sub_att.get().yaw;
+		_pub_gpos.get().eph = sqrtf(_P(X_x, X_x) + _P(X_y, X_y));
+		_pub_gpos.get().epv = sqrtf(_P(X_z, X_z));
+		_pub_gpos.get().terrain_alt = 0;
+		_pub_gpos.get().terrain_alt_valid = false;
+		if (_timeStamp - _time_last_gps < 1) {
+			_pub_gpos.get().dead_reckoning = false;
+		} else {
+			_pub_gpos.get().dead_reckoning = true;
+		}
+		_pub_gpos.update();
+	}
+}
+
+void BlockLocalPositionEstimator::publishFilteredFlow() {
+	// publish filtered flow
+	if(isfinite(_pub_filtered_flow.get().sumx) &&
+		isfinite(_pub_filtered_flow.get().sumy) &&
+		isfinite(_pub_filtered_flow.get().vx) &&
+		isfinite(_pub_filtered_flow.get().vy)) {
+		_pub_filtered_flow.update();
+	}
+}
+
+void BlockLocalPositionEstimator::predict() {
+	if (_sub_att.get().R_valid) {
+		math::Matrix<3,3> R_att(_sub_att.get().R);
+		math::Vector<3> a(_sub_sensor.get().accelerometer_m_s2);
+		_u = R_att*a;
+		_u(2) += 9.81f; // add g
+	} else {
+		_u = math::Vector<3>({0,0,0});
+	}
+
+	// dynamics matrix
+	math::Matrix<n_x, n_x>  A; // state dynamics matrix
+	// derivative of position is velocity
+	A(X_x, X_vx) = 1;
+	A(X_y, X_vy) = 1;
+	A(X_z, X_vz) = 1;
+	// derivative of velocity is accelerometer
+	// 	bias + acceleration
+	//_A(X_vx, X_bx) = 1;
+	//_A(X_vy, X_by) = 1;
+	//_A(X_vz, X_bz) = 1;
+
+	// input matrix
+	math::Matrix<n_x, n_u>  B; // input matrix
+	B(X_vx, U_ax) = 1;
+	B(X_vy, U_ay) = 1;
+	B(X_vz, U_az) = 1;
+
+	// input noise
+	math::Matrix<n_u, n_u> R;
+	R(U_ax, U_ax) =
+		_accel_xy_stddev.get()*_accel_xy_stddev.get();
+	R(U_ay, U_ay) =
+		_accel_xy_stddev.get()*_accel_xy_stddev.get();
+	R(U_az, U_az) =
+		_accel_z_stddev.get()*_accel_z_stddev.get();
+
+	// process noise matrix
+	math::Matrix<n_x, n_x>  Q; // process noise
+	float pn_p_sq = _pn_p_stddev.get()*_pn_p_stddev.get();
+	float pn_v_sq = _pn_v_stddev.get()*_pn_v_stddev.get();
+	Q(X_x, X_x) = pn_p_sq;
+	Q(X_y, X_y) = pn_p_sq;
+	Q(X_z, X_z) = pn_p_sq;
+	Q(X_vx, X_vx) = pn_v_sq;
+	Q(X_vy, X_vy) = pn_v_sq;
+	Q(X_vz, X_vz) = pn_v_sq;
+
+	// continuous time kalman filter prediction
+	_x += (A*_x + B*_u)*getDt();
+	_P += (A*_P + _P*A.transposed() +
+		B*R*B.transposed() + Q)*getDt();
+}
+
+void BlockLocalPositionEstimator::correctFlow() {
+
+	// flow measurement matrix and noise matrix
+	math::Matrix<n_y_flow, n_x> C;
+	C(Y_flow_x, X_x) = 1;
+	C(Y_flow_y, X_y) = 1;
+
+	math::Matrix<n_y_flow, n_y_flow> R;
+	R(Y_flow_x, Y_flow_x) =
+		_flow_xy_stddev.get()*_flow_xy_stddev.get();
+	R(Y_flow_y, Y_flow_y) =
+		_flow_xy_stddev.get()*_flow_xy_stddev.get();
+
+	float flow_speed[3] = {0.0f, 0.0f, 0.0f};
+	float speed[3] = {0.0f, 0.0f, 0.0f};
+	float global_speed[3] = {0.0f, 0.0f, 0.0f};
+
+	/* rotation matrix for transformation of optical flow speed vectors */
+	static const int8_t rotM_flow_sensor[3][3] =   {
```

Comment:
Actually, I think just removing this bit should do it. I changed the driver so that the rotation is applied right as the data comes in, so it's consistent system-wide.
:)